### DISCUSSION TITLE
Installing react-router-dom-v5-compat installs react-router-dom v6 (and causes type errors)

### DISCUSSION THREAD

**Discussion by dclark27** [remix-run/react-router#9448] [v5 to v6 Migration] [answered]
I installed react-router-dom-v5-compat and was immediately hit with a mountain of errors from typescript.

"location" does not exist
"render" does not exist
...etc. You know, v6 stuff! 

But I have v5 installed and only installed react-router-dom-v5-compat.

Is there something I need to do with my package lock or tsconfig to make sure it is using v5 types even though v6 is a dependency of react-router-dom-v5-compat?

Thanks!

package.json for good measure (shortened):
```
"devDependencies": {
    ...
    "@types/react": "17.0.33",
    "@types/react-dom": "17.0.9",
    "@types/react-router-dom": "5.3.2",
    "typescript": "4.8.4",
    ...
},
"dependencies": {
    ...
    "react": "17.0.2",
    "react-dom": "17.0.2",
    "react-router-dom": "5.3.4",
    "react-router-dom-v5-compat": "6.4.2",
    ...
}
```

and an actual error:

```
ERROR in ./src/components/policyholder/PolicyholderRoutes.tsx 163:9-15
TS2322: Type '{ path: any; render: () => Element; }' is not assignable to type 'IntrinsicAttributes & RouteProps'.
  Property 'render' does not exist on type 'IntrinsicAttributes & RouteProps'.
    161 |                                                       <Route
    162 |                                                               path={path}
  > 163 |                                                               render={() => <Redirect to={`${url}/overview`} />}
        |                                                               ^^^^^^
    164 |                                                       />
```

**Comment by dclark27** [comment, marked as answer]
Anyone who stumbles here, just delete node_modules and package-lock and reinstall. This fixed for me!

**Comment by mathewTH** [comment]
A better solution might be to add `@types/react-router` explicitly to the `devDependencies` in `packages.json`
https://github.com/remix-run/react-router/discussions/9892#discussioncomment-10763981
